### DISCUSSION TITLE
Autolock feature - windows app

### DISCUSSION THREAD

**Discussion by Fire-Rage** [ente-io/ente#3505] [Enhancements]
Would love to have an auto lock feature after a time duration on the Windows App. it has it on the Android app but can't do it on the PC. have to manually lock it.

**Comment by Damariobros** [comment]
Yes please

**Comment by SamB-GB** [comment]
Would like to see this in the MacOS app as well.

**Comment by Fire-Rage** [comment]
@prateekmedia Any update on this?

**Comment by etim808** [comment]
I would like to see this feature added as well. It would be much appreciated.

**Comment by Fire-Rage** [comment]
Any updates? This is a security risk and should be taken with higher priority.

**Comment by etim808** [comment]
I strongly agree. This feature is much needed and poses a great security risk.

**Comment by JoelAzaria** [comment]
If implemented please do so as an OPTION.    I would NOT want this as I use many TOTP in a day and it would be HIGHLY annoying.

If i need to secure my desktop I lock my desktop.   Different strokes for different folks...

**Comment by semyl** [comment]
I would like this too, as per #2652

**Comment by VRProject1** [comment]
Yes please!
See and read as Security Vulnerability before this functionality implementation!

**Comment by ITSSmitty870** [comment]
Would love to see this also!

**Comment by Surfict** [comment]
Hello, 

Adding my voice to the crowd! Currently, anyone can access to it if someone forget to lock his session for a few minutes. You could need Ente only once in the beginning of the day, and it would stay the entire day accessible without protection/lock. A bit worrying!

**Comment by Nyghl** [comment]
Yes, this is a much needed feature!

**Comment by CrawliestCotter** [comment]
I also would like to see this feature be added. Especially since the minimize on close feature was added to desktop, and the app doesn't lock on screen lock so it's easy for a user to forget to manually close the app.

As of now, I'd have a hard time suggesting the desktop app to my friends until these two features are added.

**Reply by CrawliestCotter** [reply]
I wanted to add that in Windows you can create a Task Scheduler Event that runs taskkill /F /T /IM "auth.exe" on a workstation lock trigger. (Just make sure to uncheck this option in the conditions menu: "start the task only if the computer is on AC power" if you're using a laptop). 

This at least means that if I walk away from my pc for long enough to automatically lock my screen the app will be closed and subsequently be in a locked state when I unlock my computer. This mostly solves the issue (though it is a little hacky). I'll have to try doing something similar with a systemd service on my linux box.

Having this feature built-in would be best though.
